public/js/actions/index.js

```diff
@@ -1,10 +1,21 @@
 "use strict";
 
+const constants = require("../constants");
 const breakpoints = require("./breakpoints");
 const eventListeners = require("./event-listeners");
 const sources = require("./sources");
 const tabs = require("./tabs");
 const pause = require("./pause");
 
+function willNavigate() {
+  return { type: constants.NAVIGATE };
+}
+
+function navigate() {
+  return sources.loadSources();
+}
+
```

Comment:
nit: a navigate const would make it more consistent.

`const navigate = { willNavigate, navigate }`